cmake/GtsamBuildTypes.cmake

```diff
@@ -9,6 +9,11 @@ function(list_append_cache var)
   set(${var} "${cur_value}" CACHE STRING "${MYVAR_DOCSTRING}" FORCE)
 endfunction()
 
+# Check for optional compiler flags, if supported:
+include(CheckCXXCompilerFlag)
+# See https://github.com/borglab/gtsam/issues/21
+check_cxx_compiler_flag(-faligned-new COMPILER_HAS_FALIGNED_NEW)
```

Comment:
What are the minimum compiler versions of gcc, clang, and VS which support this? If it's only very new compilers, I think we should still prefer to address Eigen alignment issues by appropriate placement of `EIGEN_MAKE_ALIGNED_OPERATOR_NEW` so we don't abandon users who have no control over the compiler version in their build environment.

Reply:
Agreed. Please note GTSAM is used by *many* companies on a variety of platforms, including embedded ones where compiler versions might be lagging. Incidentally, This is also why C++17 and std:shared_ptr are not suited for 4.X versions...

Reply:
FYI: I have removed this commit from the PR, since it was finally solved in a better way in #185